scantool/diag_l0_elm.c

```diff
@@ -743,13 +752,77 @@ elm_initbus(struct diag_l0_device *dl0d, struct diag_l1_initbus_args *in)
 					break;
 			}
 
+			//set init address
+			if ((dev->elmflags & ELM_323_BASIC) && (in->addr != 0x33)) {
+				fprintf(stderr, FLFMT "elm_initbus: ELM323 doesn't support target address %02X", FL, in->addr);
+				return diag_iseterr(DIAG_ERR_GENERAL);
+			} else {
+				uint8_t iia[15];
+				sprintf((char *) iia, "ATIIA %02X\x0D", in->addr);
```

Comment:
OK. Could you add an " if (in->addr != 0x33) " to only call ATIIA if necessary; the extra delay makes a difference when using the scan / probe commands.